go_lib/hooks/internal_tls/hook.go

```diff
@@ -31,6 +31,9 @@ import (
 type GenSelfSignedTLSHookConf struct {
 	// SANs - list of domains to include into certificate
 	SANs []string
+	// SANsGenerator dynamic SANs generator based in hook input (e.x.: values like cluster domain)
+	SANsGenerator func(input *go_hook.HookInput) []string
```

Comment:
I look to it and i have one idea.

We can pass  `GenSelfSignedTLSHookConf` generator (`func (*go_hook.HookInput) GenSelfSignedTLSHookConf` ) to `RegisterInternalTLSHook` function. In `RegisterInternalTLSHook` call generator and pass config to `genSelfSignedTLS`

In this case we can use only `SANs` field. It a much cleaner way than use two fields to getting SAN's.

Yep, we will change func signature, but it easy to fix in another places, because we have only 2 places to use this hook: `000-common` (it is integrate tests for internal hooks) and `301-prometheus-metrics-adapter` and in there places you only wrap configuration to function.

Reply:
It's overcomplicated. We will pass the Factory to `RegisterInternalTLSHook` and then to the handlerFunc of that hook. Then we will return a function from the returning function there. Seems like callback hell :)

My first idea was to make SANs generator like an variadic args `Option` but then I analyzed the config structure.
`CN` - is static
`Namespace` - is static
`TLSSecretName` - is static
`FullValuesPathPrefix` or `ValuesPaths` - are static

and only `SANs` could be dependable on a values input. That's why I decided to put generator near the SANS field. It's easy to observe it when you are researching the conf struct in IDE

Reply:
> Seems like callback hell 
Ok, I've accept your conclusion.

> and only SANs could be dependable on a values input. That's why I decided to put generator near the SAN's field. It's easy to observe it when you are researching the conf struct in IDE
Ok, then I suggest to remove SAN's field and stay only generator. I am confused about two way to pass SAN's and i think we need to stay only one.